src/stateMachines/maintenance/changePassword.ts

```diff
@@ -24,167 +21,247 @@ import { SecretManagerCreatorFactory } from '../../factories/SecretManagerCreato
+          input: ({ context }) => (context),
           id: 'set-on-device',
           onDone: {
-            actions: assign({ updatedPassword: (context, event) => event.data }),
+            actions: assign({ updatedPassword: ({ context, event }) => event.output as string }),
             target: 'SAVE_TO_SECRET_PROVIDER'
           },
           onError: {
             actions: assign({
-              statusMessage: (_, event) => coalesceMessage('at SET_ADMIN_ACL_ENTRY', event.data)
+              errorMessage: ({ event }) => coalesceMessage('at SET_ADMIN_ACL_ENTRY', event.error)
             }),
             target: 'FAILED'
           }
         }
       },
       SAVE_TO_SECRET_PROVIDER: {
+        entry: assign({
+          message: ({ context, event }) => '',
+          errorMessage: ({ context, event }) => ''
+        }),
         invoke: {
-          src: this.saveToSecretProvider.bind(this),
+          src: 'saveToSecretProvider',
+          input: ({ context }) => (context),
           id: 'save-to-secret-provider',
           onDone: {
             target: 'REFRESH_MPS'
           },
           onError: {
             actions: assign({
-              statusMessage: (_, event) => coalesceMessage('at SAVE_TO_SECRET_PROVIDER', event.data)
+              errorMessage: ({ event }) => coalesceMessage('at SAVE_TO_SECRET_PROVIDER', event.error)
             }),
             target: 'FAILED'
           }
         }
       },
       REFRESH_MPS: {
+        entry: assign({
+          message: ({ context, event }) => '',
+          errorMessage: ({ context, event }) => ''
+        }),
         invoke: {
-          src: this.refreshMPS.bind(this),
+          src: 'refreshMPS',
+          input: ({ context }) => (context),
           id: 'refresh-mps',
           onDone: {
             target: 'SUCCESS'
           },
           onError: {
             actions: assign({
-              statusMessage: (context, event) => coalesceMessage('at REFRESH_MPS', event.data)
+              errorMessage: ({ context, event }) => coalesceMessage('at REFRESH_MPS', event.error)
             }),
             target: 'FAILED'
           }
         }
       },
+      ERROR: {
+        entry: sendTo('error-machine', { type: 'PARSE' }),
+        invoke: {
+          src: 'error',
+          id: 'error-machine',
+          input: ({ context, event }) => ({
+            message: context.message,
+            clientId: context.clientId
+          }),
+          onError: {
+            actions: assign({ message: ({ event }) => event.error }),
+            target: 'FAILED'
+          },
+          onDone: 'NEXT_STATE'
+        },
+        on: {
+          ONFAILED: {
+            actions: assign({ errorMessage: ({ context, event }) => coalesceMessage(context.errorMessage, event.output) }),
+            target: 'FAILED'
+          }
+        }
+      },
+      NEXT_STATE: {
+        always: [
+          {
+            guard: 'isGeneralSettings',
```

Comment:
Is NEXT_STATE required?  Where does this go if isGeneralSettings is false?

Reply:
added path to FAILED state.